src/mount-support.c

```diff
@@ -293,3 +294,58 @@ void setup_slave_mount_namespace()
 		die("can not make make / rslave");
 	}
 }
+
+#define BIND_MAX_LINE_LENGTH	512	// arbitrary
+
+void setup_bind_mounts(const char *appname)
+{
+	debug("setup_bind_mounts %s", appname);
+	FILE *f = NULL;
+	const char *bind_profile_dir = "/var/lib/snapd/bind/profiles/";
+
+	char profile_path[512];	// arbitrary path name limit
+	int snprintf_rc =
+	    snprintf(profile_path, sizeof(profile_path), "%s/%s.bind",
+		     bind_profile_dir, appname);
+	if (snprintf_rc < 0 || snprintf_rc >= 512) {
+		errno = 0;
+		die("snprintf returned unexpected value");
+	}
+
+	f = fopen(profile_path, "r");
+	// it is ok for the file to not exist
+	if (f == NULL && errno == ENOENT)
+		return;
+	// however any other error is a real error
+	if (f == NULL) {
+		fprintf(stderr, "Can not open %s (%s)\n", profile_path,
```

Comment:
I guess you can just `die("cannot open %s", profile_path)` and let it do the rest